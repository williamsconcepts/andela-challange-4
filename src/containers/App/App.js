import React, { Component } from 'react';
import Header from '../../components/Header/Header';
import Form from '../../components/Form/Form';
import Display from '../../components/Display/Display';

import classes from './App.module.css';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      input: true,
      data: {
        region: {
          name: null,
          avgAge: null,
          avgDailyIncomeInUSD: null,
          avgDailyIncomePopulation: null
        },
        periodType: null,
        timeToElapse: null,
        reportedCases: null,
        population: null,
        totalHospitalBeds: null
      },
      results: {
        data: {},
        impact: {},
        severeImpact: {}
      }
    };
  }

  // eslint-disable-next-line class-methods-use-this
  getInputValues(event) {
    const name = event.target.getAttribute('name');
    const value = Number.parseFloat(event.target.value) || event.target.value;
    return { name, value };
  }

  formatNumber = (x) => (x ? x.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',') : x);

  regionalInputHandler = (event) => {
    const { name, value } = this.getInputValues(event);
    this.setState((prevState) => {
      let { data } = prevState;
      let { region } = data;
      region = { ...region, ...{ [name]: value } };
      data = { ...data, region };
      return { data };
    });
  };
  // /^\d*\.?(?:\d{1,})?$/

  otherInputHandler = (event) => {
    const { name, value } = this.getInputValues(event);
    this.setState((prevState) => {
      let { data } = prevState;
      data = { ...data, ...{ [name]: value } };
      return { data };
    });
  };

  estimate = () => {
    this.setState({ input: false });
  }

  render() {
    const { input } = this.state;
    const { results } = this.state;
    const output = input ? (
      <Form
        regionDataChanged={this.regionalInputHandler}
        otherInputChanged={this.otherInputHandler}
        clicked={this.estimate}
      />
    ) : (
      <Display format={this.formatNumber} results={results} />
    );
    return (
      <div className={classes.App}>
        <Header heading={classes.Heading} />
        {output}
      </div>
    );
  }
}

export default App;
