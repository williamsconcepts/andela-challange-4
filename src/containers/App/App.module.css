@font-face {
  font-display: swap;
  font-family: 'SEGA LOGO FONT';
  font-style: normal;
  font-weight: normal;
  src: local('SEGA LOGO FONT'), url('SEGA.woff') format('woff');
}

.Heading,
h1 {
  font-family: 'SEGA LOGO FONT';
  font-weight: 200;
  font-size: 3.5rem;
  color: #0ccac4;
}

.Loading {
  margin: 17rem auto;
}

@media (max-width: 1024px) {
  .Heading,
  h1 {
    font-size: 3rem;
  }
}

@media (max-width: 500px) {
  .Heading,
  h1 {
    font-size: 2rem;
  }
}
