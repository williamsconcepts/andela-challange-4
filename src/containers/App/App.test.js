import React from 'react';
import { shallow } from 'enzyme';
import { createSerializer } from 'enzyme-to-json';
import App from './App';

expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

describe('App Component Snapshot', () => {
  test('', () => {
    expect.assertions(1);
    expect(shallow(<App />)).toMatchSnapshot();
  });
});
