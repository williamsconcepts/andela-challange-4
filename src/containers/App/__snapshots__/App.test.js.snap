// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Snapshot  1`] = `
<div
  className="App"
>
  <header
    heading="Heading"
  />
  <display
    format={[Function]}
    results={
      Object {
        "data": Object {},
        "impact": Object {},
        "severeImpact": Object {},
      }
    }
  />
</div>
`;
